tests/unit_tests/test_read_data.py

```diff
@@ -0,0 +1,121 @@
+
+
+def return_valid_csv_data():
+    """
+    Function which returns expected data structure of csv file.
+    """
+
+    csv_data = [
+        {
+            "country": "Brazil",
+            "capital": "Brasilia",
+            "avg_temp_celsius": "21.5",
+            "annual_rainfall_mm": "1500",
+            "climate_zone": "Tropical",
+        },
+        {
+            "country": "Canada",
+            "capital": "Ottawa",
+            "avg_temp_celsius": "6.3",
+            "annual_rainfall_mm": "940",
+            "climate_zone": "Continental",
+        },
+        {
+            "country": "Egypt",
+            "capital": "Cairo",
+            "avg_temp_celsius": "22.1",
+            "annual_rainfall_mm": "25",
+            "climate_zone": "Desert",
+        },
+    ]
+    return csv_data
+
+
+def return_valid_json_data():
+    """
+    Function which returns expected data structure of json file.
+    """
+
+    json_data = {
+        "climate_data": [
+            {
+                "country": "Brazil",
+                "capital": "Brasilia",
+                "climate_info": {
+                    "avg_temp_celsius": 21.5,
+                    "annual_rainfall_mm": 1500,
+                    "climate_zone": "Tropical",
+                },
+                "natural_disasters": ["Floods", "Landslides"],
+            },
+            {
+                "country": "Canada",
+                "capital": "Ottawa",
+                "climate_info": {
+                    "avg_temp_celsius": 6.3,
+                    "annual_rainfall_mm": 940,
+                    "climate_zone": "Continental",
+                },
+                "natural_disasters": ["Blizzards", "Wildfires"],
+            },
+        ]
+    }
+    return json_data
+
+
+@pytest.mark.parametrize(
+    "filepath, expected_output",
+    (
+        (
+            os.path.join(fixtures_folder, "valid_climate_json_data.json"),
+            return_valid_json_data(),
+        ),
+        (
+            os.path.join(fixtures_folder, "valid_climate_csv_data.csv"),
+            return_valid_csv_data(),
+        ),
+    ),
+)
+def test_valid_files_return_expected_output(
+    filepath: str, expected_output: Union[dict, list[dict[str, Any]]]
+):
+    assert os.path.exists(filepath)
+    data = read_data_file(filepath)
+    assert data is not None
+    assert data == expected_output
+
+
+@pytest.mark.parametrize(
+    "filepath, expected_output",
+    (
+        (
+            os.path.join(fixtures_folder, "invalid_climate_json_data.json"),
+            return_valid_json_data(),
+        ),
+        (
+            os.path.join(fixtures_folder, "invalid_climate_csv_data.csv"),
+            return_valid_csv_data(),
+        ),
+    ),
+)
+def test_invalid_files_do_not_return_expected_output(
+    filepath: str, expected_output: Union[dict, list[dict[str, Any]]]
+):
+    assert os.path.exists(filepath)
+    data = read_data_file(filepath)
```

Comment:
Does the program not throw an error when the data given can't be read by the CSV reader/JSON parser?